src/Microsoft.TemplateEngine.Edge/Installers/NuGet/IDownloader.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.TemplateEngine.Edge.Installers.NuGet
+{
+    internal interface IDownloader
+    {
+        Task<NuGetPackageInfo> DownloadPackageAsync(string downloadPath, string identifier, string version = null, IEnumerable<string> additionalSources = null, CancellationToken cancellationToken = default);
+    }
+
+    internal struct NuGetPackageInfo
+    {
+        internal string Author;
+        internal string FullPath;
+        internal string NuGetSource;
+        internal string PackageIdentifier;
+        internal string PackageVersion;
```

Comment:
Can this be immutable?

Reply:
Converted it into immutable class, really wish we had https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/init or `records` :)